miscnn/processing/preprocessor.py

```diff
@@ -62,55 +63,31 @@ class Preprocessor:
                                                 (True), or should the subfunctions preprocessing be performed during runtime? (False).
         prepare_batches (boolean):              Should all batches be prepared and backup to disk before starting the training (True),
                                                 or should the batches be created during runtime? (False).
-        analysis (string):                      Modus selection of analysis type. Options:
```

Comment:
Wouldn't remove the analysis type out of there, due to it is still required / expected to be passed as parameter

Reply:
or?

Reply:
the parameter list does not change. Additional patching arguments are still handeled and passed along to the patch handler using variable parameters. There are multiple python libraries that use this method for wrapping objects. This avoid requiring code changes for the end user in all cases execept overlap, while allowing extension down the line without making substantial changes to the preprocessor. It would also simplify overloading the patch handler down the line i believe.

Reply:
Please rework